core/util/src/main/java/org/eclipse/rdf4j/common/iteration/IntersectIteration.java

```diff
@@ -99,6 +99,7 @@ protected boolean accept(E object) throws X {
 		return false;
 	}
 
+	// this method does not seem to "addSecondSet" since the second set seems to be ignored
 	public Set<E> addSecondSet(Iteration<? extends E, ? extends X> arg2, Set<E> set) throws X {
 		return Iterations.addAll(arg2, makeSet());
 	}
```

Comment:
This looks very suspicious. Is this method used anywhere else? I'd like to take close look at this class over the weekend.

Reply:
It was used somewhere, but not to “add a second set”. Not used anymore after I swapped that call with Iterations.asSet()